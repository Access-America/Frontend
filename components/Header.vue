<template>
  <div class="flex items-center justify-between border-b border-base-lighter">
    <h1 class="px-4 text-gray-90 font-bold text-sm"><slot></slot></h1>
    <button class="block uppercase bg-primary hover:bg-primary-dark text-white p-3 text-sm focus:outline-none focus:shadow-outline">Menu</button>
  </div>
</template>

<script>
export default {
  name: "Header"
}
</script>

<style scoped>

</style>
