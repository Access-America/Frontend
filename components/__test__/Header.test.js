import { mount } from '@vue/test-utils'
import Header from '~/components/Header.vue'

describe('Header', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(Header)
  })

  it('should be a Vue instance', () => {
    expect(wrapper.isVisible()).toBeTruthy()
  })
  it('should render properly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
