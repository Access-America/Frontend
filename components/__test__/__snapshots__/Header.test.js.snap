// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <a
    class="usa-skipnav"
    href="#main-content"
  >
    Skip to main content
  </a>
   
  <div
    class="tw-hidden"
  >
    <div
      class="usa-nav is-visible usa-overlay is-visible"
    />
  </div>
   
  <div
    class="usa-overlay"
  />
   
  <header
    class="usa-header usa-header--basic"
  >
    <div
      class="usa-nav-container"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="basic-logo"
        >
          <em
            class="usa-logo__text"
          >
            <a
              aria-label="Home"
              href="/"
              title="Home"
            />
          </em>
        </div>
         
        <button
          class="usa-menu-btn"
        >
          
          Menu
        
        </button>
      </div>
       
      <nav
        aria-label="Primary navigation"
        class="usa-nav"
      >
        <button
          class="usa-nav__close"
        >
          <img
            alt="close"
            src="~/assets/images/close.svg"
          />
        </button>
         
        <ul
          class="usa-nav__primary usa-accordion"
        >
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="basic-nav-section-one"
              aria-expanded="false"
              class="usa-accordion__button usa-nav__link"
            >
              <span>
                Voting
              </span>
            </button>
             
            <ul
              class="usa-nav__submenu"
              hidden="hidden"
              id="basic-nav-section-one"
            >
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   Your National Representatives
                </a>
              </li>
               
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   US Senators
                </a>
              </li>
               
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   Members Of US Congress
                </a>
              </li>
               
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   State Representatives
                </a>
              </li>
               
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   Local Representatives
                </a>
              </li>
            </ul>
          </li>
           
          <li
            class="usa-nav__primary-item"
          >
            <button
              aria-controls="basic-nav-section-two"
              aria-expanded="false"
              class="usa-accordion__button usa-nav__link"
            >
              <span>
                UBI
              </span>
            </button>
             
            <ul
              class="usa-nav__submenu"
              hidden="hidden"
              id="basic-nav-section-two"
            >
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   File for Universal Basic Income
                </a>
              </li>
               
              <li
                class="usa-nav__submenu-item"
              >
                <a
                  href="#"
                >
                   Connect A Bank Account
                </a>
              </li>
            </ul>
          </li>
        </ul>
         
        <form
          class="usa-search usa-search--small "
          role="search"
        >
          <label
            class="usa-sr-only"
            for="basic-search-field-small"
          >
            Search small
          </label>
           
          <input
            class="usa-input"
            id="basic-search-field-small"
            name="search"
            type="search"
          />
           
          <button
            class="usa-button"
            type="submit"
          >
            <span
              class="usa-sr-only"
            >
              Search
            </span>
          </button>
        </form>
      </nav>
    </div>
  </header>
</div>
`;
